apps/studio/src/components/connection/AddDatabaseForm.vue

```diff
@@ -0,0 +1,77 @@
+<template>
+  <form class="save-connection expand" @submit.prevent="save">
+    <h3 class="dialog-c-title">Add database</h3>
+
+    <div v-if="error" class="alert alert-danger">
+      {{error}}
+    </div>
+
+    <div class="form-group">
+      <input class="form-control" v-model="databaseName" @keydown.enter.prevent.stop="save" type="text" placeholder="Database Name">
+    </div>
+
+    <div class="form-group" v-if="charSets.length > 0">
+      <select v-model="selectedCharset">
+        <option v-for="charset in charSets" v-bind:key="charset" :selected="charset === defaultCharSet" :value="charset">{{charset}}</option>
+      </select>
+    </div>
+
+    <div class="form-group" v-if="collations.length > 0">
+      <select v-model="selectedCollation">
+        <option v-for="collation in collations" v-bind:key="collation" :value="collation">{{collation}}</option>
```

Comment:
All I can figure is the charset select was given a default to match the charset used in the server itself to make it easier to know which one to select instead of giving too many options? As far as a "default collation" though, not sure if there is one for the charset chosen

Reply:
portals have been added. I went through and put a portal around all the other modals except for the Export modal because that one was inconsistent in when it popped up. There's more to that sucker that needs some looking into, but most of the other ones are handled.